Added demo MEX functions

bpancras opened the pull request:
Here is the list of changes:
1. Added invertMex.cpp within cpp/invertMex
2. Added invertNumber.cpp to drive invertMex MEX function
3. Refactored the testTask and checkTask to use inbuilt tasks
4. Added a GH Action to build MEX functions on Linux, Windows and Mac, Package the toolbox with MEX function for all the three platforms 
5. Added test for invertNumber function

rpurser47 commented on .github/workflows/mexbuild.yml:
```diff
@@ -0,0 +1,58 @@
+        uses: actions/checkout@v4
+      - uses: actions/download-artifact@v4
+        with:
+          name: MEX_ubuntu-latest
```
Consider using the `pattern` and `merge_multitple` options to `download-artifact` to simplify use of [actions/download-artifact@v4](actions/download-artifact@v4)

rpurser47 commented on .github/workflows/mexbuild.yml:
```diff
@@ -0,0 +1,58 @@
+        uses: matlab-actions/run-build@v2
+        with:
+          tasks: mex
+      - name: Upload MEX functions
```
Don't you want to run your tests?

rpurser47 reviewed (approved):
Looks good.  There's a couple of suggestions, but they don't block this.

rpurser47 commented:
@bpancras I can merge this when you're ready.  Looks good.

bpancras commented:
- Updated the buildfile to reflect the current best practices
- Added clean task
- For 25a, migrated the packaging metadata within the Project, 24b still uses the toolbox packaging prj
- Added a note in the README, indicating the toolbox works well in 24b and later
